### DISCUSSION TITLE
Entity Relation disambiguation: Who *helps* and who *is helped*?

### DISCUSSION THREAD

**Discussion by metalaureate** [explosion/spaCy#9034] [Help: Model Advice] [answered]
Thank you for bringing the Transformer revolution to the masses! Do you think it is possible in spaCy to detect agent relations, helper and helpee?

John saved Sam from drowning. (John=helper, Sam=helpee)
Sam did John’s homework. (John=helpee, Sam=helper)

The variety seems endless…: eg

John went to the store for Sam. (John=helper, Sam=helpee)
Sam visited John’s mom when he went out of town to check in on her. (John=helpee, Sam=helper)
John took Sam’s dogs for a walk. (John=helper, Sam=helpee)

What NLP problem domain am I getting into here?  Is this an relation extraction problem? e.g. https://www.youtube.com/watch?v=8HL-Ap5_Axo It wasn't clear to me that it falls under relation extraction.

**Comment by polm** [comment, marked as answer]
Please don't @ individuals to get their attention, it's disruptive.

What you are describing is called "Semantic Role Labeling". Our standard advice is to avoid it unless you're very sure you need it because it's a hard problem (and you often don't need it). 

One simple approach is to use NER to identify candidates (you can assume helper/helpee are both PERSON), and then use dependencies (possibly via a DependencyMatcher) to match on common structures, like a specific verb. This has limited coverage but high precision and is easy to iterate on.

For an overview of the research check Chapter 19 of [Speech and Language Processing](https://web.stanford.edu/~jurafsky/slp3/) by Jurafsky and Martin.

**Reply by metalaureate** [reply]
Thank you for the direction. Roger that on @ (and apologies).

**Reply by metalaureate** [reply]
I'd mark this as the answer but github is bugging out when I try (422 error, unprocessable entity).

**Reply by polm** [reply]
Huh, weird, thanks for letting us know. Discussions are pretty new and sometimes they seem to get stuck still.
